### ISSUE TITLE
[BUG] Enabling log channel causes php memory exhausted

### ISSUE THREAD

**Issue by BoGnY** [nutgram/nutgram#767]
### Nutgram version

4.27.0

### PHP version

8.3.10

### Framework

Laravel 11

### Cache driver

Laravel File

### OS

Windows 11

### Dev environment

Docker

### Describe the bug

When adding `TELEGRAM_LOG_CHANNEL=telegram` on env file, php get error:
```log
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes)
```

My application is a basic app, it has only Article model with 8 attributes and about 10 records on db, nothing else.

Visiting homepage, which is `Route::get('/', view('homepage'));` and `homepage.blade.php` is empty, will throw error..

### To Reproduce

1. Add `TELEGRAM_LOG_CHANNEL=telegram` on env file
2. Visit a page
3. Get 500 error
4. Got to php log file and found memory exhausted
5. Disable log channel and all work as expected

### Expected behavior

Don't get memory exhausted

**Comment by Lukasss93** [comment]
Can you please provides a code example that reproduces the issue?

**Comment by BoGnY** [comment]
> Can you please provide a code example that reproduces the issue?

Just run:
```bash
composer create-project laravel/laravel nutgram
cd nutgram
composer require nutgram/laravel
yarn install
yarn build
```

add this to `.env`:
```env
TELEGRAM_TOKEN="api-telegram-token"
TELEGRAM_LOG_CHANNEL=telegram
NUTGRAM_LOG_CHAT_ID=channel-id
```

add this to `config/logging.php`:
```php
'telegram' => [
    'driver' => 'custom',
    'via' => \Nutgram\Laravel\Log\NutgramLogger::class,
    'level' => 'debug',
    'chat_id' => env('NUTGRAM_LOG_CHAT_ID'), // any chat_id where bot can write messages
],

'nutgram' => [
    'driver' => 'monolog',
    'level' => env('LOG_LEVEL', 'debug'),
    'handler' => StreamHandler::class,
    'formatter' => \Nutgram\Laravel\Log\NutgramFormatter::class,
    'with' => [
        'stream' => 'php://stderr',
    ],
    'processors' => [PsrLogMessageProcessor::class],
],
```

Nothing else, very basic installation without any other changes..
Visiting homepage (so the default Laravel homepage):
![image](https://github.com/user-attachments/assets/05995250-ff0b-4098-a783-8d22946eaf8b)

and I see this in php log on docker machine:
```log
[01-Oct-2024 14:04:35 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in /shared/httpd/nutgram/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 1294
[01-Oct-2024 14:04:35 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in /shared/httpd/nutgram/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php on line 256
```

**Comment by Lukasss93** [comment]
Can you paste here the output of the following command?
```sh
php -i
```

Also, can you share your docker configuration with me?

**Comment by BoGnY** [comment]
> Can you paste here the output of the following command?
> 
> ```shell
> php -i
> ```

here the output: [phpinfo.log](https://github.com/user-attachments/files/17212715/phpinfo.log)

> Also, can you share your docker configuration with me?

I use https://github.com/cytopia/devilbox environment with https://hub.docker.com/u/johnea updated images instead the original one..



But the problem exists even if I try a php-cli command on my Windows 11 host machine (php 8.3.8):
![image](https://github.com/user-attachments/assets/6659ece2-d950-426c-9b2f-a0e22021fc44)

phpinfo host machine: [phpinfo.log](https://github.com/user-attachments/files/17213018/phpinfo.log)

**Comment by Lukasss93** [comment]
This could be a problem of your dev environment. 
Try to change your dev environment to something else and let me know.

**Comment by Lukasss93** [comment]
I cannot reproduce this problem. 
Can you provide a public repository to reproduce this issue?

**Comment by BoGnY** [comment]
https://github.com/BoGnY/nutgram-oom

Add your `TELEGRAM_TOKEN` and `NUTGRAM_LOG_CHAT_ID` in `.env` file, and run a simple `php artisan list`

**Comment by Lukasss93** [comment]
> [!NOTE]
> **TL;DR:** `TELEGRAM_LOG_CHANNEL` must be any laravel channel **EXCEPT** `telegram`.

Ah! Now I see the problem.
There is a misunderstanding about the use of logs.
Here is an explanation of how to use the 2 types of logs: https://nutgram.dev/docs/configuration/laravel#logging

### TELEGRAM_LOG_CHANNEL
The environment variable `TELEGRAM_LOG_CHANNEL` is **ONLY** for logging requests and responses to/from a telegram.
It is automatically used by Nutgram only if it is filled (by default it is `null`) and **MUST NOT** be `telegram`. 
(I will explain why later).

### NUTGRAM_LOG_CHAT_ID
The environment variable `NUTGRAM_LOG_CHAT_ID` is used to explicitly log any information manually with the following code: 
```php
Log::channel('message')->info('oops');
```

### Why do I get the "php memory exhausted" error?
Because Laravel loads the Nutgram instance from its provider, which in turn loads the class that implements the LogInterface, which in turn loads the Nutgram instance again, causing it to loop and fill up memory.

### A picture is worth a thousand words
![immagine](https://github.com/user-attachments/assets/7a212841-df9a-4f95-9ab1-df6c1e461c3b)

### P.S.
*Anyway, I will make a small fix that will avoid this problem if anyone tries to trigger this problem.*
